src/main/scala/extractor/Serializables.scala

```diff
@@ -2,85 +2,102 @@ package extractor
 
 import scala.meta._
 
-final case class ImplicitParam(ctx: SemanticCtx, symbol: Symbol, denot: Denotation) extends CSV.Serializable {
-  lazy val id: String = s"${symbol.syntax}"
-  // If there are no names on the denotation, we have an implicit which defines
-  // its own type - Most likely an object
-  val name: String = denot.name.toString
-  val clazz: String = denot.names.headOption match {
-    case Some(n) => n.symbol.toString
-    case None => symbol.toString
-  }
-  val typee: String = denot.names.headOption match {
-    case Some(n) => denot.signature.toString
-    case None => name
-  }
-  val kind: String = ctx.getKind(denot)
-
-  override val csvHeader: Seq[String] = Seq("fqn", "fqfn", "fqparamlist", "fqtn", "type", "kind", "name")
-  override val csvValues: Seq[String] = Seq(id, "", "", clazz, typee, kind, name)
-
-}
-
-final case class AppTerm(term: Term, params: Int, nameEnd: Int)
-final case class FunApply(ctx: SemanticCtx, app: AppTerm, file: String) extends CSV.Serializable {
-  lazy val id: String = s"$file:$line:$col"
-
-  // Take end line and cols because function call chains have the same start
-  val line: String = app.term.pos.endLine.toString
-  val col: String = app.term.pos.endColumn.toString
-  val symbol: String = ctx.qualifiedName(app.term)
-  val code: String = app.term.toString
-  val nargs = app.params.toString
-
-  override val csvHeader: Seq[String] = Seq("sourcelink", "path", "line", "col", "code", "symbol", "fqfn", "fqparamlist", "nargs")
-  override val csvValues: Seq[String] = Seq(id, file, line, col, code, symbol, "", "", nargs)
+trait ResultElement {
+  def id: String
 }
 
-final case class SyntheticApply(ctx: SemanticCtx, synth: Synthetic, file: String, params: Int) extends CSV.Serializable {
-  lazy val id: String = s"$file:$line:$col"
-
-  // Take end line and cols because function call chains have the same start
-  val line: String = synth.position.endLine.toString
-  val col: String = synth.position.endColumn.toString
-  val symbol: String = synth.names(1).symbol.toString
-  val code: String = s"apply(${if (params > 0) { "_" + ",_" * (params - 1) }})"
-
-  override val csvHeader: Seq[String] = Seq("sourcelink", "path", "line", "col", "code", "symbol", "fqfn", "fqparamlist", "nargs")
-  override val csvValues: Seq[String] = Seq(id, file, line, col, code, symbol, "", "", params.toString)
+final case class Location(path: String, line: String, col: String) {
```

Comment:
Why is line and col a String?

Reply:
Because we need to clone them in order to not keep too many references to the tree, it was a performance issue once. Of course it can be cloned into Ints, but I guessed that for the time it reaches the case class we shouldn't perform any operation on the data anyways, so it didn't really matter.